[flaky] 3.40% tests failed in repo eventing on 2020-07-08 10:53:12 +0000 UTC

### Auto-generated issue tracking flakiness of test
* **Test name**: 3.40% tests failed in repo eventing on 2020-07-08 10:53:12 +0000 UTC
* **Repository name**: eventing

<!-------------End of issue body, Please don't edit below this line------------->
<!--[DONT_MODIFY_TEST_IDENTIFIER]3.40% tests failed in repo eventing on 2020-07-08 10:53:12 +0000 UTC[DONT_MODIFY_TEST_IDENTIFIER]-->

I think this is actually the root cause of the failures. Seems like this was the root cause, and caused all the other tests to fail if I read the logs correctly:
https://prow.knative.dev/view/gcs/knative-prow/logs/ci-knative-eventing-continuous/1280756772721659905

```
--- FAIL: TestBrokerChannelFlowTriggerV1BrokerV1 (411.62s)
    --- FAIL: TestBrokerChannelFlowTriggerV1BrokerV1/InMemoryChannel-messaging.knative.dev/v1 (30.22s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0xc002042290 addr=0x57c275 pc=0xd2fc66]

goroutine 3424 [running]:
testing.tRunner.func1.1(0x246cf00, 0x3b92e60)
	/root/.gvm/gos/go1.14.2/src/testing/testing.go:940 +0x421
testing.tRunner.func1(0xc001714000)
	/root/.gvm/gos/go1.14.2/src/testing/testing.go:943 +0x600
panic(0x246cf00, 0x3b92e60)
	/root/.gvm/gos/go1.14.2/src/runtime/panic.go:975 +0x3e3
knative.dev/eventing/test/lib/duck.GetAddressableURI(0x2b57140, 0xc000724778, 0xc0020438d8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/prow/go/src/knative.dev/eventing/test/lib/duck/resource_inspectors.go:42 +0x158
knative.dev/eventing/test/lib.(*Client).GetAddressableURI(0xc0009c0090, 0x26e4a24, 0x19, 0xc001828340, 0x3, 0x3, 0xc001f5e480, 0x1a)
	/home/prow/go/src/knative.dev/eventing/test/lib/operation.go:53 +0x17c
knative.dev/eventing/test/e2e/helpers.BrokerChannelFlowWithTransformation.func1(0xc001714000, 0x7ffef8332e2a, 0xf, 0x7ffef8332e11, 0x18)
	/home/prow/go/src/knative.dev/eventing/test/e2e/helpers/broker_channel_flow_helper.go:156 +0x990
knative.dev/eventing/test/lib.(*ComponentsTestRunner).RunTests.func1(0xc001714000)
	/home/prow/go/src/knative.dev/eventing/test/lib/test_runner.go:70 +0x98
testing.tRunner(0xc001714000, 0xc001828320)
	/root/.gvm/gos/go1.14.2/src/testing/testing.go:991 +0x1ec
created by testing.(*T).Run
	/root/.gvm/gos/go1.14.2/src/testing/testing.go:1042 +0x661
FAIL	knative.dev/eventing/test/e2e	681.080s
```

--- a/test/lib/duck/resource_inspectors.go
+++ b/test/lib/duck/resource_inspectors.go
@@ -39,6 +39,10 @@ func GetAddressableURI(dynamicClient dynamic.Interface, obj *resources.MetaResou
 
 	at := untyped.(*duckv1alpha1.AddressableType)
 
+	if at.Status.Address == nil {
+		return url.URL{}, fmt.Errorf("addressable does not have an Address: %+v", at)
+	}
+
 	au := at.Status.Address.GetURL()
 	if au.Host == "" {
 		return url.URL{}, fmt.Errorf("addressable's URL does not have a Host: %+v", at)

